app/src/main/java/app/grapheneos/camera/capturer/ImageSaver.kt

```diff
@@ -337,7 +345,10 @@ class ImageSaver(
         val imageCaptureCallbackExecutor = Executors.newSingleThreadExecutor()
         private val imageWriterExecutor = Executors.newSingleThreadExecutor()
 
+        private var photosInQueue = 0
+
         private const val TAG = "ImageSaver"
+        private const val MAX_PHOTOS_IN_QUEUE = 20
```

Comment:
20 is way too high of a limit. Depending on resolution, it can require from around 100 MiB (5 MiB per ~12 Mpx image) to close to 200 MiB of memory.
I think 5 is enough.

Reply:
What do you mean?
In vast majority of cases, actual queue length will never exceed 2, because image writeback usually is much faster than image capture.
This check is needed only for rare case of slow storage / slow SAF storage provider.
Setting this value to 10 will mean that the user can lose up to 10 last captured images in case app crashes / storage provider becomes unavailable / storage becomes completely full / etc

Reply:
Yes, we had introduced this change mainly for slow SAFs and other disk storages.

> Setting this value to 10 will mean that the user can lose up to 10 last captured images in case app crashes / storage provider becomes unavailable / storage becomes completely full / etc.

Losing the captured photos could be a good reason to reduce the limit to 5 over being able to take multiple shots (over 10) in case of slow storage providers.

Will make the required changes soon. Thanks for the feedback @muhomorr!